src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt

```diff
@@ -146,12 +146,23 @@ class VisitService(
       booking.addApplication(application)
     }
 
-    with(application.visitContact!!) {
-      booking.visitContact = VisitContact(visit = booking, visitId = booking.id, name = name, telephone = telephone)
+    application.visitContact?.let {
+      booking.visitContact?.let { visitContact ->
+        visitContact.name = it.name
+        visitContact.telephone = it.telephone
+      } ?: run {
+        booking.visitContact = VisitContact(
+          visit = booking,
+          visitId = booking.id,
+          name = it.name,
+          telephone = it.telephone,
+        )
+      }
     }
 
     application.support.let {
       booking.support.clear()
+      visitRepository.saveAndFlush(booking)
```

Comment:
are we doing  save here but not in line 150 because we clear it out?

Reply:
yes